src/FsAutoComplete/CodeFixes/AdjustConstant.fs

```diff
@@ -0,0 +1,1517 @@
+        | SynConst.Char    _ -> Some <| nameof(System.Char)
+        | SynConst.Byte    _ -> Some <| nameof(System.Byte)
+        | SynConst.SByte   _ -> Some <| nameof(System.SByte)
+        | SynConst.Int16   _ -> Some <| nameof(System.Int16)
+        | SynConst.UInt16  _ -> Some <| nameof(System.UInt16)
+        | SynConst.Int32   _ -> Some <| nameof(System.Int32)
+        | SynConst.UInt32  _ -> Some <| nameof(System.UInt32)
+        | SynConst.Int64   _ -> Some <| nameof(System.Int64)
+        | SynConst.UInt64  _ -> Some <| nameof(System.UInt64)
+        | SynConst.IntPtr  _ -> Some <| nameof(System.IntPtr)
+        | SynConst.UIntPtr _ -> Some <| nameof(System.UIntPtr)
+        | SynConst.Single  _ -> Some <| nameof(System.Single)
+        | SynConst.Double  _ -> Some <| nameof(System.Double)
+        | SynConst.Decimal _ -> Some <| nameof(System.Decimal)
+        | _ -> None
+
+      let isSystemAssembly (assembly: FSharpAssembly) =
+        match assembly.SimpleName with
+        // dotnet core
+        | "System.Runtime"
+        // .net framework
+        | "mscorlib"
+        // .net standard
+        | "netstandard"
+          -> true
+        | _ -> false
+
+      let assemblies = parseAndCheck.GetCheckResults.ProjectContext.GetReferencedAssemblies()
+      let ty =
+        assemblies
+        |> Seq.filter (isSystemAssembly)
+        |> Seq.tryPick (fun system -> system.Contents.FindEntityByPath ["System"; name])
+        |> Option.map (fun ent -> ent.AsType())
+
+      // Note: `ty` should never be `None`: we're only looking up standard dotnet types -- which should always be available.
+      //       But `isSystemAssembly` might not handle all possible assemblies with default types -> keep it safe and return `option`
+
+      return (name, ty)
+    }
+  /// Fix that replaces `constantRange` with `propertyName` on type of `constant`.
+  ///
+  /// Example:
+  /// `constant = SynConst.Double _` and `fieldName = "MinValue"`
+  /// -> replaces `constantRange` with `Double.MinValue`
+  ///
+  /// Tries to detect if leading `System.` is necessary (`System` is not `open`).
+  /// If cannot detect: Puts `System.` in front
+  let replaceWithNamedConstantFix
+    doc
+    (pos: FcsPos) (lineStr: String)
+    (parseAndCheck: ParseAndCheckResults)
+    (constant: SynConst)
+    (constantRange: Range)
+    (fieldName: string)
+    (mkTitle: string -> string)
+    = option {
+        let! (tyName, ty) = tryGetFSharpType parseAndCheck constant
+        let propCall =
+          ty
+          |> Option.bind (fun ty ->
+            parseAndCheck.GetCheckResults.GetDisplayContextForPos pos
+            |> Option.map (fun displayContext -> $"{ty.Format displayContext}.{fieldName}")
+          )
+          |> Option.defaultWith (fun _ -> $"System.{tyName}.{fieldName}")
+        let title = mkTitle $"{tyName}.{fieldName}"
+        let edits = [| { Range = constantRange; NewText = propCall } |]
```

Comment:
Is there an easier way to detect if `System` is `open`? Or Format some code string at a certain location?

All I want is: `Format 'System.Int.MaxValue' but only put System in front if necessary`.  

What I currently do: Find "`System`" Assembly by looking through all Assemblies and then get `System.Int` to use for formatting (with `DisplayContext`).
... That's rather ... involved ...